developers/command-api.rst

```diff
@@ -0,0 +1,169 @@
+The Command API
+==========================
+
+What better way to expand your plugins using commands? This page will show how
+to create, register and unregister commmands.
+
+
+Create the command class
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Each command class implements the `Command interface <https://github.com/VelocityPowered/Velocity/blob/master/api/src/main/java/com/velocitypowered/api/command/Command.java>`_, which has two methods:
+one for when the command is executed and one to provide suggestions for tab completion.
+Let's see an example of a simple command that will tell whoever executes the command
+"Hello World" in light blue text.
+
+.. code-block:: java
+
+    package com.example.velocityplugin;
+
+    import com.velocitypowered.api.command.Command;
+    import com.velocitypowered.api.command.CommandSource;
+    import net.kyori.text.TextComponent;
+    import net.kyori.text.format.TextColor;
+    import org.checkerframework.checker.nullness.qual.NonNull;
+
+    public class CommandTest implements Command {
+
+        @Override
+        public void execute(@NonNull CommandSource source, String[] args) {
+            source.sendMessage(TextComponent.of("Hello World!").color(TextColor.AQUA));
+        }
+    }
+
+Now that we have created the command we need to register it in order for it to work.
+We have a `Command Manager <https://github.com/VelocityPowered/Velocity/blob/master/api/src/main/java/com/velocitypowered/api/command/CommandManager.java>`_ where we can register our commands.
+We get the command manager by executing ``proxyServer.getCommandManager()`` with 
+the proxy instance, or by injecting it using the ``@Inject`` annotation in our
+main class. The register method requires two parameters, the command object and 
+the command aliases which is a varargs parameter.
+
+.. code-block:: java
+
+    commandManager.register(new CommandTest(), "test");
+
+If we assemble it all into our main class created on the first tutorial, it'll look
+something like this
+
+.. code-block:: java
+
+    package com.example.velocityplugin;
+
+    import com.google.inject.Inject;
+    import com.velocitypowered.api.command.CommandManager;
+    import com.velocitypowered.api.plugin.Plugin;
+    import org.slf4j.Logger;
+
+    @Plugin(id = "myfirstplugin", name = "My First Plugin", version = "1.0-SNAPSHOT",
+            description = "I did it!", authors = {"Me"})
+    public class VelocityTest {
+
+        @Inject private VelocityTest(CommandManager commandManager, Logger logger) {
+            commandManager.register(new CommandTest(), "test");
+            logger.info("Plugin has enabled!");
+        }
+    }
+
+As you can see we're injecting the commandManager instance but we can also obtain
```

Comment:
We could use a better explanation of what gets injected.

Reply:
I feel it should have a link to another page where the injection is explained more detailed. No point in explaining it here. (eg. "See how injection works <link>")

The page would explain all the interfaces that can be injected.